workflow/executor/k8sapi/client.go

```diff
@@ -92,3 +93,37 @@ func (c *k8sAPIClient) KillContainer(pod *corev1.Pod, container *corev1.Containe
 func (c *k8sAPIClient) killGracefully(ctx context.Context, containerID string) error {
 	return execcommon.KillGracefully(ctx, c, containerID)
 }
+
+func (c *k8sAPIClient) until(ctx context.Context, f func(pod *corev1.Pod) bool) error {
+	podInterface := c.clientset.CoreV1().Pods(c.namespace)
+	for {
+		done, err := func() (bool, error) {
+			w, err := podInterface.Watch(ctx, metav1.ListOptions{FieldSelector: "metadata.name=" + c.podName})
```

Comment:
Seems like the watch is still here?

Reply:
well, I guess this should be called "remove need for watch, except for k8sapi" :)